pkg/check/chunkrepair/chunkrepair.go

```diff
@@ -0,0 +1,281 @@
+
+		// download again to see if the chunk is repaired
+		data3, err := nodeC.DownloadChunk(ctx, chunk.Address(), "")
+		if err != nil { // this time it should succeed
+			return err
+		}
+		if !bytes.Equal(data3, chunk.Data()) {
+			return errors.New("chunk downloaded in NodeC does not have proper data")
+		}
+		fmt.Println("repaired chunk ", chunk.Address().String())
+
+		if pushMetrics {
+			if err := pusher.Push(); err != nil {
+				fmt.Printf("node %d: %s\n", i, err)
+			}
+		}
+	}
+	return nil
+}
+
+// getNodes get three nodes A, B, C and a chunk such that
+// NodeA's and NodeC's first byte of the address does not match
+// nodeB is the closest to the generated chunk in the cluster.
+func getNodes(ctx context.Context, c bee.Cluster, rnd *rand.Rand) (bee.Node, bee.Node, bee.Node, *bee.Chunk, error) {
+	var overlayA swarm.Address
+	var overlayB swarm.Address
+	var overlayC swarm.Address
+	var chunk *bee.Chunk
+
+	// get overlay addresses of the cluster
+	overlays, err := c.Overlays(ctx)
+	if err != nil {
+		return bee.Node{}, bee.Node{}, bee.Node{}, nil, err
+	}
+
+	// find node A and C, such that they have the greatest distance between them in the cluster
+	overlayA, overlayC, err = findFarthestNodes(overlays)
+	if err != nil {
+		return bee.Node{}, bee.Node{}, bee.Node{}, nil, err
+	}
+
+	// find node B
+	// generate a chunk and pick the closest address from all the available addresses
+	for {
+		closestOverlay, c, err := getRandomChunkAndClosestNode(overlays, rnd)
+		if err != nil {
+			return bee.Node{}, bee.Node{}, bee.Node{}, nil, err
+		}
+		if bytes.Equal(closestOverlay.Bytes(), overlayA.Bytes()) {
+			continue
+		}
+		if bytes.Equal(closestOverlay.Bytes(), overlayC.Bytes()) {
+			continue
+		}
+		// we found our chunk and closest node
+		overlayB = closestOverlay
+		chunk = c
+		break
+	}
+	fmt.Printf("overlayA: %s\n", overlayA.String())
+	fmt.Printf("overlayB: %s\n", overlayB.String())
+	fmt.Printf("overlayC: %s\n", overlayC.String())
+	fmt.Printf("chunk Address: %s\n", chunk.Address().String())
+
+	// get the nodes for all the addresses
+	var nodeA bee.Node
+	var nodeB bee.Node
+	var nodeC bee.Node
+	for _, node := range c.Nodes {
+		addresses, err := node.Addresses(ctx)
+		if err != nil {
+			return bee.Node{}, bee.Node{}, bee.Node{}, nil, err
+		}
+
+		if addresses.Overlay.Equal(overlayA) {
+			nodeA = node
+		}
+		if addresses.Overlay.Equal(overlayB) {
+			nodeB = node
+		}
+		if addresses.Overlay.Equal(overlayC) {
+			nodeC = node
+		}
+	}
+	return nodeA, nodeB, nodeC, chunk, nil
+}
+
+// uploadAndPinChunkToNode uploads a given chunk to a given node and pins it.
+func uploadAndPinChunkToNode(ctx context.Context, node *bee.Node, chunk *bee.Chunk) error {
+	err := node.UploadChunk(ctx, chunk)
+	if err != nil {
+		return err
+	}
+	pinned, err := node.PinChunk(ctx, chunk.Address())
+	if err != nil {
+		return err
+	}
+	if !pinned {
+		return errors.New("could not pin chunk in nodeA")
```

Comment:
`nodeA` is too specific for this function, it can be removed from the error message.